apps/prairielearn/src/tests/sprocs/check_assessment_access.test.ts

```diff
@@ -130,24 +132,26 @@ describe('sproc check_assessment_access* tests', function () {
 
   describe('with PrairieTest', function () {
     describe('without checked-in reservation', () => {
-      it('should allow access to an exam without exam_uuid', async () => {
+      it('should not allow access to an exam without exam_uuid', async () => {
         const authorized = await checkAssessmentAccess({
           assessment_id: '10',
           authz_mode: 'Exam',
+          authz_mode_reason: 'PrairieTest',
           course_role: 'None',
           course_instance_role: 'None',
           user_id: '1000',
           uid: 'valid@example.com',
           date: '2010-07-07 06:06:06-00',
           display_timezone: 'US/Central',
         });
-        assert.isTrue(authorized);
+        assert.isFalse(authorized);
```

Comment:
This was a change in behavior. Is it desirable? I think so! My guess is that this test meant to test an Exam-mode assessment that wasn't linked to a PT exam at all. We could always add another test for that? Though I'm pretty sure this is already covered by the "without PrairieTest" block above.

Reply:
`authz_mode_reason` comes from `authzCourseOrInstance` via `ip_to_mode`; this is entirely unrelated to whether or not a particular access rule we're checking has `exam_uuid` or not. This test specifically tests the case where someone is in a CBTF with a PT reservation and they try to access an assessment without an `exam_uuid`.

That said: this is all complex, so don't necessarily take my assertions up there at face value. Does this sound correct to you?

Reply:
I buy your logic!